### PR TITLE
Include detailed style notes on commits, rebases, PRs

### PR DESCRIPTION
**Pull request by onebree**
- Pulled information from our private wiki, and thoughtbot's public guides
- Placed in the style directory, rather than protocol, since:
  + Protocol is being dropped in another pr (#30)
  + This information is still helpful for folks viewing our public guide (like new devs or applicants)

### CODE REVIEW

File: style/git/README.md [hunk 1]
```diff
@@ -1,9 +1,71 @@
...
+
+- You may want to `fixup` or `squash` commits if you are correcting typos or making small fixes to a single class frequently.
+- You may want to use `reword` if you prefixed a commit message with `WIP`, when it is actually an isolated change.
+- Or, if you are refactoring a class, you may want that in its own commit, avoiding the rebase method altogether!
+
+As with everything, use your best judgement to rebase.
+
+## Make small PRs
+
+PRs with over 250 lines of additive diff are considered large. _It is valuable to artificially break up PRs for the sake of making them smaller._ With that said, always try to make PRs deployable on their own (i.e. introduce a feature flag if needed). The smaller the PR, the easier it is to review, the faster it is to get it deployed.
```
Comment (joofsh) [inline style/git/README.md]: love this 👍 Small PRs for the win 🙌

File: style/git/README.md [hunk 2]
```diff
@@ -1,9 +1,71 @@
...
+git rebase -i origin/master
+git rebase -i HEAD~5
+git rebase -i release/feature-work
+```
+
+- You may want to `fixup` or `squash` commits if you are correcting typos or making small fixes to a single class frequently.
+- You may want to use `reword` if you prefixed a commit message with `WIP`, when it is actually an isolated change.
+- Or, if you are refactoring a class, you may want that in its own commit, avoiding the rebase method altogether!
+
+As with everything, use your best judgement to rebase.
```
Comment (joofsh) [inline style/git/README.md]: Do we want to potentially provide more guidance on when to rebase? Assuming this is in fact our culture, something like "Generally speaking, you should aim to rebase and squash commits down to cohesive groups of code changes". Or potentially "Each commit that will go onto master should be stable, working code"

**Review by joofsh — APPROVED** [review]
few comments, but generally looks great! 👍 🙏

File: style/git/README.md (hunk 2 above)
Comment (onebree) [reply style/git/README.md]: I like both of those suggestions! I will add them :-)

**Pull request merged**